iree/compiler/Conversion/LinalgToSPIRV/ConvertToSPIRVPass.cpp

```diff
@@ -131,9 +131,15 @@ Value getPushConstantValue(Operation *op, unsigned elementCount,
 spirv::GlobalVariableOp getOrInsertResourceVariable(Location loc, Type type,
                                                     unsigned set,
                                                     unsigned binding,
+                                                    StringRef funcName,
                                                     Block &block) {
-  auto name = llvm::formatv("__resource_var_{0}_{1}__", set, binding).str();
+  auto name =
+      llvm::formatv("__resource_var_{0}_{1}_{2}__", set, binding, funcName)
+          .str();
   for (auto varOp : block.getOps<spirv::GlobalVariableOp>()) {
+    // TODO(#3573): Attach Aliased decoration to spv.globalVariable when having
+    // overlapping set/binding numbers in shaders
+    if (varOp.type() != type) return nullptr;
```

Comment:
Should this be an assert instead? Is it possible that the same resource has different type in a shader? Looking at the code this would just cause a crash later.